operators/constellation-node-operator/internal/cloud/azure/client/scalinggroup.go

```diff
@@ -165,7 +165,7 @@ func imageReferenceFromImage(img string) (*armcompute.ImageReference, error) {
 	}
 
 	// expecting image to not be a marketplace image
-	if strings.HasPrefix(img, "/CommunityGalleries") {
+	if strings.HasPrefix(img, "/communityGalleries") {
```

Comment:
Can we make this a case insensitive compare? This way, the code still works if Azure decides to change the API under our feet.

Reply:
Is there even a use case for the `else` part anymore?

Reply:
I think we only have community images and market images so we should be able to remove it.

Reply:
I'm pretty sure that future security types on Azure will have restrictions regarding the use of community galleries. This was already the case for SEV-SNP in the past.

Reply:
And then we'd need to resort to private images? Maybe it's just not clear to me what one would use the third option for.

Reply:
That's correct in simplified terms. The `ID` field is used for most image references that are not community (public to the world) or direct shared. Azure has a lot of other types of images and sharing that are less convenient.

Reply:
For instance, there is this alternative sharing option that still uses the `ID` field: https://learn.microsoft.com/en-us/azure/virtual-machines/share-gallery